src/core/qgsvectordataprovider.h

```diff
@@ -93,6 +94,7 @@ class CORE_EXPORT QgsVectorDataProvider : public QgsDataProvider, public QgsFeat
       WriteLayerMetadata = 1 << 22, //!< Provider can write layer metadata to the data store. Since QGIS 3.0. See QgsDataProvider::writeLayerMetadata()
       CancelSupport = 1 << 23, //!< Supports interruption of pending queries from a separated thread. Since QGIS 3.2
       CreateRenderer = 1 << 24, //!< Provider can create feature renderers using backend-specific formatting information. Since QGIS 3.2. See QgsVectorDataProvider::createRenderer().
+      CreateLabeling = 1 << 24, //!< Provider can set labeling settings using backend-specific formatting information. Since QGIS 3.4. See QgsVectorDataProvider::createLabeling().
```

Comment:
```suggestion
      CreateLabeling = 1 << 25, //!< Provider can set labeling settings using backend-specific formatting information. Since QGIS 3.4. See QgsVectorDataProvider::createLabeling().
```

Reply:
We should eventually get rid of all those hardcoded numbers, what are they useful for?

Reply:
Which numbers? The bit shift operators?

Reply:
Forget about my comment: the values must be assigned anyway (C++ has its own glitches).